src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -330,6 +330,78 @@ namespace OpenMS
       return integer_data_arrays_;
     }
 
+    /// Returns an iterator to the first float data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename FloatDataArrays::iterator getFloatDataArrayByName(const String& name)
+    {
+      typename FloatDataArrays::iterator it = float_data_arrays_.begin();
+      for (; it != float_data_arrays_.end(); ++it)
+      {
+        if (it->getName() == name) return it;
+      }
+      return it;
+    }
+
+    /// Returns an iterator to the first string data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename StringDataArrays::iterator getStringDataArrayByName(const String& name)
```

Comment:
hmm I am pretty bad in getting templated code right on the first try but I think you probably mean something like
```
template <class T>
typename T::StringDataArray::iterator T::getStringDataArrayByName(const String& name)
{...}
in a common header of MSSpectrum / MSChromatogram (+- some additional typename - I usually get those wrong ;)